conda_smithy/configure_feedstock.py

```diff
@@ -520,6 +520,40 @@ def _collapse_subpackage_variants(
         "BUILD",
     }
 
+    # on osx, merge MACOSX_DEPLOYMENT_TARGET & c_stdlib_version to max of either; see #1884
+    if squished_used_variants["target_platform"][0].startswith("osx"):
```

Comment:
I'm confused. Don't we have to check each entry in the list, not just the first to see if osx is used anywhere?

Reply:
I don't have much experience with this code (e.g. what's in `squished_used_variants` etc. and how to use it). From what I could tell, in "regular" recipes, each arch gets a separate run here, being populated only with a single `target_platform`. Perhaps that's just the way it happens in the tests here though.

Reply:
As soon as lists longer than one are involved, one needs to resolve the zip_keys (again, AFAICT), and it looks very hard to come up with a sane config. That's why I'm basically throwing my hands up if anything unexpected comes along (i.e. if the list has length > 1; I could extend that check to `squished_used_variants["target_platform"]`)

Reply:
What I mean by that is that there are more corner cases (e.g. if a feedstock builds for several different `MACOSX_DEPLOYMENT_TARGET`) where the warning is upgraded to an error to avoid over-complicating the merge logic. I think this is a fair trade-off, but please let me know if you disagree.

Reply:
If there's a test cases where `squished_used_variants["target_platform"]` ends up being list longer than 1, I'd be happy to dig deeper.

Reply:
Ahhhhh ok. I think if you move this code up a bit in the function and have it work on each dict in the set `all_variants`, then you will always have one element for each variable and it should work for edge cases too.

Reply:
Also, we should be able to handle zip keys there too by inspecting them for each variant and adding `MACOSX_DEP...` to the group if `c_stdlib_version` shows up in it.

Reply:
> if you move this code up a bit in the function and have it work on each dict in the set `all_variants` [...]

Thanks, that was really helpful! 🙏